tests/grpc/test_utils.go

```diff
@@ -0,0 +1,75 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	pGRPC "get.porter.sh/porter/gen/proto/go/porterapis/porter/v1alpha1"
+	"get.porter.sh/porter/pkg/config"
+	pServer "get.porter.sh/porter/pkg/grpc/portergrpc"
+	"get.porter.sh/porter/pkg/porter"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+
+	//"go.opentelemetry.io/otel/attribute"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+type TestPorterGRPCServer struct {
+	TestPorter       *porter.TestPorter
+	TestPorterConfig *config.TestConfig
+}
+
+const bufSize = 1024 * 1024
+
+var lis *bufconn.Listener
+
+func NewTestGRPCServer(t *testing.T) (*TestPorterGRPCServer, error) {
+	srv := &TestPorterGRPCServer{
+		TestPorter:       porter.NewTestPorter(t),
+		TestPorterConfig: config.NewTestConfig(t),
+	}
+	return srv, nil
+}
+
+func (s *TestPorterGRPCServer) newTestInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
```

Comment:
any possible way to be more specific than the `interface{}` param and return types?

Reply:
The interface{} is used by the GRPC milddleware.  Fixing the type here could cause issues.